recmetrics/metrics.py

```diff
@@ -310,6 +388,10 @@ def make_confusion_matrix(y: list, yhat: list) -> None:
     plt.yticks([0,1], [1,0])
     plt.show()
 
+def _precision(predicted, actual):
```

Comment:
Let me know if this refactor is okay! I pulled `_precision` out of `recommender_precision` so that I could reuse it in the `_apk` calc. I also moved the rounding back inside of `recommender_precision` as to not have breaking changes there but also so that the raw precision scores aren't rounded.

If you'd prefer I didn't do this, I can re-implement this precision calc inside of `_apk`